scripts/folder_diff.ts

```diff
@@ -41,30 +42,60 @@ if (!fs.existsSync(outputFolder)) {
   fs.mkdirSync(outputFolder);
 }
 
-// Loop through files in folder1
-fs.readdirSync(folder1)
-  .filter((file) => file.endsWith(".png"))
-  .forEach((file) => {
-    const basefile = basename(file);
-
-    // Check if file exists in folder2
-    if (fs.existsSync(`${folder2}/${basefile}`)) {
-      // Compare the files
-      try {
-        execSync(`cmp -s "${folder1}/${basefile}" "${folder2}/${basefile}"`);
-      } catch (e) {
-        // If files are different
-        fs.copyFileSync(
+// Function to compare two images using resemble.js
+async function compareImages(
+  image1Path: string,
+  image2Path: string
+): Promise<boolean> {
+  return new Promise((resolve, reject) => {
+    const image1 = fs.readFileSync(image1Path);
+    const image2 = fs.readFileSync(image2Path);
+
+    resemble(image1)
+      .compareTo(image2)
+      .onComplete((data) => {
+        // Consider images different if there's any mismatch
+        // misMatchPercentage of 0 means images are identical
+        resolve(data.misMatchPercentage > 0);
```

Comment:
Good, we have the ability to increase this threshold in the future in case anything turns out to be slightly machine-dependent or nondeterministic.